src/ros_filter.cpp

```diff
@@ -687,8 +687,8 @@ namespace RobotLocalization
     filter_.setSensorTimeout(sensorTimeout);
 
     // Set the timeToPublish timer
-    timeToPublish_.setFrequency(frequency_);
     timeToPublish_.setConditionVariable(measurementsReady_);
```

Comment:
I added a sleep between `setFrequency` and `setConditionVariable` (before swapping them) to reproduce the problem. It was very easy to reproduce it that way.